client/packages/common/src/ui/components/inputs/TextInput/NumbericTextInput.test.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import { NumericTextInput, NumericTextInputProps } from './NumericTextInput';
+import { TestingProvider } from '@common/utils';
+
+describe('Test NumericTextInput component', () => {
+  const TestNumericTextInput = (props: NumericTextInputProps) => {
+    const [value, setValue] = useState<number | undefined>();
+    return (
+      <TestingProvider>
+        <NumericTextInput
+          value={value}
+          onChange={value => setValue(value)}
+          {...props}
+        />
+      </TestingProvider>
+    );
+  };
+
+  it('should handle decimal number input', async () => {
+    const { getByRole } = render(<TestNumericTextInput decimalLimit={10} />);
+    const input = getByRole('textbox');
+
+    fireEvent.change(input, { target: { value: '1.' } });
+    expect(input).toHaveValue('1.');
+    fireEvent.blur(input);
+    expect(input).toHaveValue('1');
+
+    fireEvent.change(input, { target: { value: '1.7001' } });
+    expect(input).toHaveValue('1.7001');
+    fireEvent.blur(input);
+    expect(input).toHaveValue('1.7001');
+
+    fireEvent.change(input, { target: { value: '1.700' } });
+    expect(input).toHaveValue('1.700');
+    fireEvent.blur(input);
+    expect(input).toHaveValue('1.7');
+  });
+
+  it('should handle non-negative decimal number input', async () => {
+    const { getByRole } = render(
+      <TestNumericTextInput decimalLimit={10} allowNegative={false} />
```

Comment:
Because a simple toggle is nicer to read. You never need to write `allowNegative={false}` since that's the default. So when you want to allow negatives you just use:
```js
<NumericTextInput allownegative />
```
which is cleaner and much easier to understand than always having to go:
```js
<NumericTextInput min={-Infinity} />
```
or
```js
<NumericTextInput min={-NumUtils.MAX_SAFE_API_INTEGER} />
```

Yes, you can achieve the same thing two ways, but I'm always in favour of offering a simple option for common use cases and saving the more verbose/obscure syntax for when you really need it.

Reply:
was thinking about allowNegative in combination of something min={-20} but this is not really a problem. Thanks for clarifying

Reply:
Yeah, `min` takes priority over `allowNegative` in the event they conflict.